diff --git a/dist/cli/index.mjs b/dist/cli/index.mjs
index 6b164e87ea8d16e3ff238d9d60ee77edefaa1096..8c1d5b7118e603da29f3a320ea9d31ae88605088 100755
--- a/dist/cli/index.mjs
+++ b/dist/cli/index.mjs
@@ -6803,7 +6803,7 @@ const updatePackageJson = async (builtUiPath) => {
 	for (const [dependency, version] of Object.entries(packageJson.devDependencies)) packageJson.devDependencies[dependency] = version.replace("workspace:", "");
 	await fs.promises.writeFile(packageJsonPath, JSON.stringify(packageJson), "utf8");
 };
-const build$1 = async ({ dir: emailsDirRelativePath, packageManager }) => {
+const build$1 = async ({ dir: emailsDirRelativePath, packageManager, prepare }) => {
 	if (packageManager) console.warn("The --packageManager option is deprecated and ignored. The build command now just uses npm.");
 	try {
 		const usersProjectLocation = process.cwd();
@@ -6842,6 +6842,7 @@ const build$1 = async ({ dir: emailsDirRelativePath, packageManager }) => {
 		await forceSSGForEmailPreviews(emailsDirPath, builtPreviewAppPath);
 		spinner.setText("Updating package.json's build and start scripts");
 		await updatePackageJson(builtPreviewAppPath);
+		if (prepare) { stopSpinnerAndPersist(spinner, { text: "Prepared preview app", symbol: logSymbols.success }); return; }
 		if (!isInReactEmailMonorepo) {
 			spinner.setText("Installing dependencies on `.react-email`");
 			const previousInclude = process.env.NPM_CONFIG_INCLUDE;
@@ -7793,7 +7794,7 @@ if (!requiredFlags.every((flag) => process.execArgv.includes(flag))) spawn(proce
 else {
 	program.name("react-email").description("A live preview of your emails right in your browser").version(package_default.version);
 	program.command("dev").description("Starts the preview email development app").option("-d, --dir <path>", "Directory with your email templates", "./emails").option("-p --port <port>", "Port to run dev server on", "3000").option("-c, --clients <clients>", "Comma-separated list of email clients to show compatibility warnings for (overrides COMPATIBILITY_EMAIL_CLIENTS)", parseClientsOption).action(dev);
-	program.command("build").description("Copies the preview app for onto .react-email and builds it").option("-d, --dir <path>", "Directory with your email templates", "./emails").addOption(new Option("-p, --packageManager <name>").hideHelp()).action(build$1);
+	program.command("build").option("--prepare", "Only prepare the preview app").description("Copies the preview app for onto .react-email and builds it").option("-d, --dir <path>", "Directory with your email templates", "./emails").addOption(new Option("-p, --packageManager <name>").hideHelp()).action(build$1);
 	program.command("start").description("Runs the built preview app that is inside of \".react-email\"").action(start);
 	program.command("export").description("Build the templates to the `out` directory").option("--outDir <path>", "Output directory", "out").option("-p, --pretty", "Pretty print the output", false).option("-t, --plainText", "Set output format as plain text", false).option("-d, --dir <path>", "Directory with your email templates", "./emails").option("-e, --extension <extension>", "Set a custom file extension for rendered emails (e.g. blade.php)").option("-s, --silent", "To, or not to show a spinner with process information", false).action(({ outDir, pretty, plainText, silent, dir: srcDir, extension }) => exportTemplates(outDir, srcDir, {
 		silent,
